File: app/src/main/java/com/leodeleon/popmovies/data/RepositoryFactory.java
package com.leodeleon.popmovies.data;

import com.leodeleon.popmovies.data.local.MovieDB;
import com.leodeleon.popmovies.data.local.PopMoviesDB;
import com.leodeleon.popmovies.data.remote.MovieAPI;
import javax.inject.Inject;
import retrofit2.Retrofit;

public class RepositoryFactory {

  private Retrofit retrofit;
  private PopMoviesDB popMoviesDB;

  @Inject
  public RepositoryFactory(Retrofit retrofit, PopMoviesDB popMoviesDB) {
    this.retrofit = retrofit;
    this.popMoviesDB = popMoviesDB;
  }

  public MovieAPI createMovieAPI() {
    return new MovieAPI(retrofit);
  }

  public MovieDB createMovieDB() {
    return new MovieDB(popMoviesDB);
  }

}

File: app/src/main/java/com/leodeleon/popmovies/data/RepositoryFactory.kt
package com.leodeleon.popmovies.data

import com.leodeleon.popmovies.data.local.MovieDB
import com.leodeleon.popmovies.data.local.PopMoviesDB
import com.leodeleon.popmovies.data.remote.MovieAPI
import retrofit2.Retrofit
import javax.inject.Inject

class RepositoryFactory @Inject
constructor(private val retrofit: Retrofit, private val popMoviesDB: PopMoviesDB) {

  fun createMovieAPI(): MovieAPI {
    return MovieAPI(retrofit)
  }

  fun createMovieDB(): MovieDB {
    return MovieDB(popMoviesDB)
  }

}
